app/app.go

```diff
@@ -182,5 +170,15 @@ func removeDatabase(cfg *config.Config) error {
 func openDB(cfg *config.Config) (database.Database, error) {
 	dbPath := databasePath(cfg)
 	log.Infof("Loading database from '%s'", dbPath)
-	return ldb.NewLevelDB(dbPath, leveldbCacheSizeMiB)
+	db, err := ldb.NewLevelDB(dbPath, leveldbCacheSizeMiB)
+	if err != nil {
+		return nil, err
+	}
+
+	err = checkDatabaseVersion(dbPath)
+	if err != nil {
+		return nil, err
+	}
+
+	return db, nil
```

Comment:
Why after opening the db?
I think this should be done before (because theoretically we might support multiple different dbs and the error should be more correct)

Reply:
Hmmmm... it was to avoid checking if the folder exists when creating the file etc. 
But yeah, you are right, we might fail to open because of bad version.

We should do the check before open, and the write after open.